docs/source/API/core/view/realloc.rst

```diff
@@ -0,0 +1,156 @@
+		const size_t n3 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n4 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n5 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n6 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n7 = KOKKOS_IMPL_CTOR_DEFAULT_ARG);
+
+* Resizes ``v`` to have the new dimensions without preserving its contents.
+
+  - ``v``: existing view, can be a default constructed one.
+
+  - ``n[X]``: new length for extent X.
+
+* Restrictions: ``View<T, P...>::array_layout`` is either ``LayoutLeft`` or ``LayoutRight``.
+
+.. code-block:: cpp
+
+   template <class I, class T, class... P>
+   void realloc(const I& arg_prop, Kokkos::View<T, P...>& v,
+		const size_t n0 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n1 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n2 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n3 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n4 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n5 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n6 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n7 = KOKKOS_IMPL_CTOR_DEFAULT_ARG);
+
+* Resizes ``v`` to have the new dimensions without preserving its contents.
+  The new ``Kokkos::View`` is constructed using the View constructor property ``arg_prop``, e.g., Kokkos::WithoutInitializing.
+
+  - ``v``: existing view, can be a default constructed one.
+
+  - ``n[X]``: new length for extent X.
+
+  - ``arg_prop``: View constructor property, e.g., ``Kokkos::WithoutInitializing``.
+
+
+* Restrictions: ``View<T, P...>::array_layout`` is either ``LayoutLeft`` or ``LayoutRight``.
+
+.. code-block:: cpp
+
+   template <class... ViewCtorArgs, class T, class... P>
+   void realloc(const Impl::ViewCtorProp<ViewCtorArgs...>& arg_prop,
+		Kokkos::View<T, P...>& v,
+		const size_t n0 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n1 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n2 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n3 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n4 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n5 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n6 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+		const size_t n7 = KOKKOS_IMPL_CTOR_DEFAULT_ARG);
+
+* Resizes ``v`` to have the new dimensions without preserving its contents.
+  The new ``Kokkos::View`` is constructed using the View constructor properties ``arg_prop``, e.g., ``Kokkos::view_alloc(Kokkos::WithoutInitializing)``.
+
+  - ``v``: existing view, can be a default constructed one.
+
+  - ``n[X]``: new length for extent X.
+
+  - ``arg_prop``: View constructor properties, e.g., ``Kokkos::view_alloc(Kokkos::WithoutInitializing)``.
+
+
+* Restrictions:
+
+  - ``View<T, P...>::array_layout`` is either ``LayoutLeft`` or ``LayoutRight``.
+
+  - ``arg_prop`` must not include a pointer to memory, a label, or a memory space.
+
+.. code-block:: cpp
+
+   template <class T, class... P>
+   void realloc(Kokkos::View<T, P...>& v,
+		const typename Kokkos::View<T, P...>::array_layout& layout);
+
```

Comment:
There's a lot of redundancy that can be eliminated.  Why is "* Resizes ``v`` to have the new dimensions without preserving its contents." repeated?  This sentence can be moved to the Description and said once.

Reply:
we are documenting each method. IMO each method should be able to stand on its own.